### PR TITLE
feat : gw7905 implement action button to rename and delete bookmark folder

### PR DESCRIPTION
**Pull request by mudana-grune**
https://youtrack.weseek.co.jp/issue/GW-7905
### Test Result

https://user-images.githubusercontent.com/92426728/201890935-a9891c8c-6931-474d-8ad9-8d82b4ac6635.mp4

### CODE REVIEW

File: packages/app/src/components/Sidebar/Bookmarks/BookmarkFolderItem.tsx [hunk 1]
```diff
@@ -39,40 +46,50 @@ const BookmarkFolderItem: FC<BookmarkFolderItemProps> = (props: BookmarkFolderIt
...
+        setIsRenameAction(false);
+        toastSuccess(t('Rename Bookmark Folder Success'));
+      }
+      else {
+        await apiv3Post('/bookmark-folder', { name: folderName, parent: targetFolder });
+        setIsOpen(true);
+        setIsRenameInputShown(false);
+        mutateChildBookmarkData();
+        toastSuccess(t('Create New Bookmark Folder Success'));
+      }
```
Comment (kaoritokashiki) [inline packages/app/src/components/Sidebar/Bookmarks/BookmarkFolderItem.tsx:86]: Could you describe this code?
I'd like to know when this implement is executed.

File: packages/app/src/components/Sidebar/Bookmarks/DeleteBookmarkFolderModal.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  const { t } = useTranslation();
+  const {
+    isOpen, onClickDeleteButton, bookmarkFolder, onModalClose,
+  } = props;
+
+  return (
+    <Modal size="md" isOpen={isOpen} toggle={onModalClose} data-testid="page-delete-modal" className="grw-create-page">
+      <ModalHeader tag="h4" toggle={onModalClose} className="bg-danger text-light">
+        <i className="icon-fw icon-trash"></i>
+        {t('Delete Bookmark Folder')}
```
Comment (kaoritokashiki) [inline packages/app/src/components/Sidebar/Bookmarks/DeleteBookmarkFolderModal.tsx:29]: please apply i18n 🙏

File: packages/app/src/components/Sidebar/Bookmarks/DeleteBookmarkFolderModal.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      </ModalHeader>
+      <ModalBody>
+        <div className="form-group pb-1">
+          <label>{ t('Delete this bookmark folder and its contents') }:</label><br />
+          <FolderIcon isOpen={false}/> {bookmarkFolder.name}
+        </div>
+        {t('Deleted folder and its contents cannot be recovered')}
```
Comment (kaoritokashiki) [inline packages/app/src/components/Sidebar/Bookmarks/DeleteBookmarkFolderModal.tsx:36]: please apply i18n

File: packages/app/src/components/Sidebar/Bookmarks/DeleteBookmarkFolderModal.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      </ModalBody>
+      <ModalFooter>
+        <button
+          type="button"
+          className="btn btn-danger"
+          onClick={onClickDeleteButton}
+        >
+          <i className="mr-1 icon-trash" aria-hidden="true"></i>
+          {t('Delete Folder')}
```
Comment (kaoritokashiki) [inline packages/app/src/components/Sidebar/Bookmarks/DeleteBookmarkFolderModal.tsx:45]: please apply i18n

**Review by kaoritokashiki — CHANGES_REQUESTED** [review]

File: packages/app/src/components/Sidebar/Bookmarks/BookmarkFolderItem.tsx (hunk 1 above)
Comment (mudana-grune) [reply packages/app/src/components/Sidebar/Bookmarks/BookmarkFolderItem.tsx:86]: It is executed when creating a new folder or a subfolder of an existing folder (Plus button). Whereas the other lines above it are for rename . Create a new folder and rename the folder uses the same component

**Review by kaoritokashiki — CHANGES_REQUESTED** [review]

File: packages/app/src/components/Sidebar/Bookmarks/BookmarkContents.tsx [hunk 5]
```diff
@@ -11,55 +11,63 @@ import BookmarkFolderNameInput from './BookmarkFolderNameInput';
...
 
   const { t } = useTranslation();
-  const [isRenameInputShown, setIsRenameInputShown] = useState<boolean>(false);
+  const [isCreateAction, setIsCreateAction] = useState<boolean>(false);
   const { mutate: mutateChildBookmarkData } = useSWRxBookamrkFolderAndChild(null);
 
 
-  const onClickBookmarkFolder = () => {
-    setIsRenameInputShown(true);
+  const onClickNewBookmarkFolder = () => {
```
Comment (kaoritokashiki) [inline packages/app/src/components/Sidebar/Bookmarks/BookmarkContents.tsx:21]: please use `useCallback` (same as other methods)

**Review by kaoritokashiki — CHANGES_REQUESTED** [review]

File: packages/app/src/components/Sidebar/Bookmarks/BookmarkContents.tsx [hunk 6]
```diff
@@ -11,55 +11,63 @@ import BookmarkFolderNameInput from './BookmarkFolderNameInput';
...
+      toastSuccess(t('toaster.create_succeeded', { target: t('bookmark_folder.bookmark_folder') }));
     }
     catch (err) {
       toastError(err);
     }
 
   }, [mutateChildBookmarkData, t]);
 
-  return (
+  const renderAddNewBookmarkFolder = () => (
```
Comment (kaoritokashiki) [inline packages/app/src/components/Sidebar/Bookmarks/BookmarkContents.tsx:39]: please use `useCallback`

**Review by kaoritokashiki — CHANGES_REQUESTED** [review]

**Review by kaoritokashiki — APPROVED** [review]

**Pull request merged**